sympy/functions/combinatorial/numbers.py

```diff
@@ -1400,6 +1401,181 @@ def _eval_is_positive(self):
+    >>> motzkin(23)
+    1129760415
+    >>> motzkin(12.132)
+    17673.6827206284 + 0.00229270467798908*I
+
+    >>> motzkin.find_first_n_motzkin(15)
+    [1, 1, 2, 4, 9, 21, 51, 127, 323, 835, 2188, 5798, 15511, 41835, 113634]
+    >>> motzkin.find_motzkin_numbers_in_range(50, 100000)
+    [51, 127, 323, 835, 2188, 5798, 15511, 41835]
+    >>> motzkin.is_motzkin(3192727797)
+    24
+    >>> motzkin.is_motzkin(486431896)
+    False
+    >>> motzkin.is_prime(953467954114363)
+    True
+
+    >>> from sympy import Symbol, Sum, binomial, hyper
+    >>> n = Symbol('n')
+
+    >>> motzkin(n).rewrite(Sum)
+    Sum(binomial(n, 2*_k)*catalan(_k), (_k, 0, floor(n/2)))
+
+    >>> motzkin(n).rewrite(binomial)
+    Sum(binomial(2*_k, _k)*binomial(n, 2*_k)/(_k + 1), (_k, 0, floor(n/2)))
+
+    >>> motzkin(n).rewrite(hyper)
+    hyper((0.5 - 0.5*n, -0.5*n), (2,), 4)
+
+    See Also
+    ========
+
+    bell, bernoulli, binomial, catalan, euler, fibonacci, tribonacci, harmonic, lucas,
+    genocchi, partition, sympy.functions.special.hyper.hyper
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Motzkin_number
+    .. [2] https://mathworld.wolfram.com/MotzkinNumber.html
+    .. [3] https://oeis.org/A001006
+
+    """
+
+    @classmethod
+    def eval(cls, n):
+        n = sympify(n)
+        from sympy.functions.special.hyper import hyper
+        if not n.is_symbol:
+            m = hyper((0.5*(1 - n), -0.5*n), [2], 4).evalf()
+            if n.is_Integer and n.is_nonnegative:
+                return int(m)
+            else:
+                return m
+
+    @staticmethod
+    def find_first_n_motzkin(n):
+        n = sympify(n)
+        motzkins = []
+        if n.is_symbol:
+            raise TypeError('Expected number, not %s'%(type(n)))
+        for i in range(int(n)):
+            motzkins.append(motzkin(i))
+        return motzkins
+
+    @staticmethod
+    def is_motzkin(n):
+        n = sympify(n)
+        if n.is_symbol:
+            raise TypeError('Expected number, not %s'%(type(n)))
+        if n.is_Integer and n.is_positive:
+            i = 0
+            num = motzkin(i)
+            while num <= n:
+                if num == n:
+                    return i
+                i += 1
+                num = motzkin(i)
+            return False
+        else:
+            return False
+
+    @staticmethod
+    def find_motzkin_numbers_in_range(x, y):
+        x, y = sympify([x, y])
+        if x.is_symbol:
+            raise TypeError('Expected number, not %s'%(type(x)))
+        if  y.is_symbol:
+            raise TypeError('Expected number, not %s'%(type(y)))
+        x = ceiling(x)
+        y = floor(y)
+        motzkin_in_range = []
+        i = 0
+        num = motzkin(i)
+        while num <= y:
+            if num >= x:
+                motzkin_in_range.append(num)
+            i += 1
+            num = motzkin(i)
+        return motzkin_in_range
+
+    @staticmethod
+    def is_prime(n):
```

Comment:
The name `is_prime` conflicts with the same name in the assumptions system. Perhaps it would be better to just implement `_eval_is_prime` here (and make motzkin not evaluate by default for very large input integers).

Reply:
Although honestly, this could also just be a note in the docstring.

Reply:
I'll just add this to the docstring